ISSUE-4 Fix timezone warning upon submit success

sigee opened the pull request:
Need to check whether it is working or not, because lots of things are changed.

lewismc commented:
Hi @sigee I just tried your new PR, thank you for submitting it here.
I get the following
```

Warning: include_once(/Users/lmcgibbn/Sites/admin/sitepod/sitepod/inc../vendor/autoload.php): failed to open stream: No such file or directory in /Users/lmcgibbn/Sites/admin/sitepod/sitepod/inc/startup.php on line 33

Warning: include_once(): Failed opening '/Users/lmcgibbn/Sites/admin/sitepod/sitepod/inc../vendor/autoload.php' for inclusion (include_path='.::./inc:./inc/functions:./inc/classes:./settings:./templates') in /Users/lmcgibbn/Sites/admin/sitepod/sitepod/inc/startup.php on line 33

Warning: session_start(): Cannot send session cache limiter - headers already sent (output started at /Users/lmcgibbn/Sites/admin/sitepod/sitepod/inc/startup.php:33) in /Users/lmcgibbn/Sites/admin/sitepod/sitepod/inc/functions/functions.inc.php on line 103

Fatal error: Class 'Sitepod\LayoutEngine' not found in /Users/lmcgibbn/Sites/admin/sitepod/sitepod/inc/functions/functions.inc.php on line 106
```
I'll debug tomorrow... something is definitely up.

sigee commented:
In the **inc/startup.php** at line 33 I left a slash character.
The included path would be
'/Users/lmcgibbn/Sites/admin/sitepod/sitepod/inc../vendor/autoload.php'
instead of
 '/Users/lmcgibbn/Sites/admin/sitepod/sitepod/inc/../vendor/autoload.php'

Oh, and I forgot to mention that it needs composer to generate the autoloading file.
There was a composer.json in the project, so I extended it with autoloading.
So you need to generate the autoload files.
[Composer](https://getcomposer.org/) sould be installed on your computer.
After that you can generate the autoloading files with it.
See here: https://getcomposer.org/doc/03-cli.md#dump-autoload 
```bash
composer dump-autoload
```
This command should generate a **vendor** folder with the depencencies and the autoloading files into the project root folder.

lewismc commented:
Excellent @sigee, I've updated all of the documentation with the above and tested locally. Seems to be working nicely however after executing a submit from the index.php I see the following Fatal Error 
```
Successful: Settings written to file settings/settings.inc.php!
Info: Scanning filesystem for files now...

Fatal error: Uncaught exception 'Exception' with message 'DateTimeZone::__construct(): Unknown or bad timezone ()' in /Users/lmcgibbn/Sites/admin/sitepod/sitepod/inc/functions/functions.inc.php:233 Stack trace: #0 /Users/lmcgibbn/Sites/admin/sitepod/sitepod/inc/functions/functions.inc.php(233): DateTimeZone->__construct('') #1 /Users/lmcgibbn/Sites/admin/sitepod/sitepod/inc/functions/parsing.inc.php(366): getDateTimeISO(1481876308) #2 /Users/lmcgibbn/Sites/admin/sitepod/sitepod/inc/functions/parsing.inc.php(208): handleURL('https://podaac....', 1481876308) #3 /Users/lmcgibbn/Sites/admin/sitepod/sitepod/inc/functions/parsing.inc.php(63): runFilesystemHandler(Array, Array) #4 /Users/lmcgibbn/Sites/admin/sitepod/sitepod/index.php(35): parseFilesystem() #5 {main} thrown in /Users/lmcgibbn/Sites/admin/sitepod/sitepod/inc/functions/functions.inc.php on line 233
```
If you want I can take a screenshot as well here.

The above relates to the following line of code
```
$timeZone = new DateTimeZone($geoData['time_zone']);
```

sigee commented:
@lewismc,
I fixed that error too. It was because of localhost is not a valid IP address, that you can use against geoip.
I added a default value in case of geoip returns empty string as time_zone.

lewismc commented:
This looks much better @sigee thank you, I see the following upon submit
```
Successful: Settings written to file settings/settings.inc.php!
Info: Loaded 125 files from file cache.
Info: Scanning filesystem for files now...
Info: Found 125 files on local filesystem.
Info: Timeout occured
Please click here to continue scanning your site.
Info: Number of files that are in the todo list: 0
Found 125 files
```
Note the ```Info: Timeout occured```

sigee commented:
@lewismc,
It is because of the "Scan website - will scan the website with the build-in crawler engine; might take some time if there are many urls to crawl! Optional: crawl only this url:" is checked in. I can't say more info about that at the moment, but I've checked the master, that does the same.
And if I run it without the "Scan website" it finishes immediately.
Does it make sense?

lewismc commented:
+1
Can you please merge this in to master branch?
Thank you VERY much, this is a much more comprehensive solution than the
one I provided. I added you are an admin so you have push rights.
Thanks

On Fri, May 26, 2017 at 8:01 AM Dávid Szigecsán <notifications@github.com>
wrote:

> @lewismc <https://github.com/lewismc>,
> It is because of the "Scan website - will scan the website with the
> build-in crawler engine; might take some time if there are many urls to
> crawl! Optional: crawl only this url:" is checked in. I can't say more info
> about that at the moment, but I've checked the master, that does the same.
> And if I run it without the "Scan website" it finishes immediately.
> Does it make sense?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/nasa/sitepod/pull/12#issuecomment-304305921>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABHJl8fDWnkoC-j7LXq5KpsIcKDvo3jfks5r9uk9gaJpZM4NnBWf>
> .
>
-- 

*Lewis*
Dr. Lewis J. McGibbney Ph.D, B.Sc
Director, MCMA Associates
Phone: +1(626)498-3090
Skype: lewis.john.mcgibbney
Email: lewis.mcgibbney@gmail.com

sigee commented:
@lewismc,
Thanks.
Yes, I got write access, but I don't want to merge this one.
As far as I know, the merge is done by other person. Never the same as the PR / change is created by.
